src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -0,0 +1,337 @@
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => _typeParameters;
+
+        public override bool IsExtensionMethod
+        {
+            get
+            {
+                // It is an error to be an extension method, but we need to compute it to report it
+                var firstParam = _syntax.ParameterList.Parameters.FirstOrDefault();
+                return firstParam != null &&
+                    !firstParam.IsArgList &&
+                    firstParam.Modifiers.Any(SyntaxKind.ThisKeyword);
+            }
+        }
+
+        internal override TypeSymbol IteratorElementType
+        {
+            get
+            {
+                return _iteratorElementType;
+            }
+            set
+            {
+                Debug.Assert((object)_iteratorElementType == null || _iteratorElementType == value);
+                Interlocked.CompareExchange(ref _iteratorElementType, value, null);
+            }
+        }
+
+        public override MethodKind MethodKind => MethodKind.LocalFunction;
+
+        public sealed override Symbol ContainingSymbol => _containingSymbol;
+
+        public override string Name => _syntax.Identifier.ValueText;
+
+        public SyntaxToken NameToken => _syntax.Identifier;
+
+        internal override bool HasSpecialName => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+
+
+        public override ImmutableArray<Location> Locations => ImmutableArray.Create(_syntax.Identifier.GetLocation());
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray.Create(_syntax.GetReference());
+
+
+        internal override bool GenerateDebugInfo => true;
+
+        public override ImmutableArray<CustomModifier> ReturnTypeCustomModifiers => ImmutableArray<CustomModifier>.Empty;
+
+        internal override MethodImplAttributes ImplementationAttributes => default(MethodImplAttributes);
+
+        internal override ObsoleteAttributeData ObsoleteAttributeData => null;
+
+
+        internal override MarshalPseudoCustomAttributeData ReturnValueMarshallingInformation => null;
+
+        internal override CallingConvention CallingConvention => CallingConvention.Default;
+
+        internal override bool HasDeclarativeSecurity => false;
+        internal override bool RequiresSecurityObject => false;
+
+        public override Symbol AssociatedSymbol => null;
+
+        public override Accessibility DeclaredAccessibility => ModifierUtils.EffectiveAccessibility(_declarationModifiers);
+
+        public override bool IsAsync => (_declarationModifiers & DeclarationModifiers.Async) != 0;
+
+        public override bool IsStatic => (_declarationModifiers & DeclarationModifiers.Static) != 0;
+
+        public override bool IsVirtual => (_declarationModifiers & DeclarationModifiers.Virtual) != 0;
```

Comment:
Some of these we do not intend to support (e.g. `virtual`). It should be an error if they arise. On the other hand, once the error has been reported the API should probably still report that the symbol is `virtual`.